File: leetcode/medium/Q49_GroupAnagrams.py
from typing import List, Dict

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        answer = []
        dictionary = {}
        for i in range(len(strs)):
            tmp_map = [0] * 26
            for c in strs[i]:
                tmp_map[ord(c) - 97] += 1
            if str(tmp_map) in dictionary:
                dictionary[str(tmp_map)].append(strs[i])
            else:
                dictionary[str(tmp_map)] = [strs[i]]

        for d in dictionary.values():
            answer.append(d)
        return answer

File: src/medium/Q49_GroupAnagrams.java
package medium;

import java.util.*;

public class Q49_GroupAnagrams {
    class Solution {
        public List<List<String>> groupAnagrams(String[] strs) {
            Map<String, List<String>> wordMap = new HashMap<>();
            for(String s : strs){
                char[] temp = s.toCharArray();
                Arrays.sort(temp);
                String key = String.valueOf(temp);
                if(!wordMap.containsKey(key)){
                    wordMap.put(key, new ArrayList<>());
                }
                wordMap.get(key).add(s);
            }
            List<List<String>> answer = new ArrayList<>();
            for(Map.Entry<String, List<String>> entry : wordMap.entrySet()){
                answer.add(entry.getValue());
            }
            return answer;
        }
    }
}
